tests/validation/tests/v3_api/test_bkp_restore_s3_recover.py

```diff
@@ -0,0 +1,169 @@
+                     "/name.html")
+
+    # Cleanup task
+    if RANCHER_CLEANUP_CLUSTER:
+        for node in etcd_nodes+master_node:
+            ips_to_remove.append(node.externalIpAddress)
+            client.delete(node)
+        for ip_to_remove in ips_to_remove:
+            delete_node_from_ec2(ip_to_remove)
+
+
+@pytest.fixture(scope='module', autouse=True)
+def create_project_client_and_cluster_s3_three_etcd(request):
+    node_roles = [
+        ["controlplane", "worker"],
+        ["etcd"], ["etcd"], ["etcd"],
+    ]
+    cluster, aws_nodes = create_and_validate_custom_host(node_roles)
+    client = get_user_client()
+
+    p, ns = create_project_and_ns(USER_TOKEN, cluster, "testrecover")
+    p_client = get_project_client_for_token(p, USER_TOKEN)
+    c_client = get_cluster_client_for_token(cluster, USER_TOKEN)
+
+    namespace["p_client"] = p_client
+    namespace["ns"] = ns
+    namespace["cluster"] = cluster
+    namespace["project"] = p
+    namespace["c_client"] = c_client
+    namespace["client"] = client
+
+    def fin():
+        cluster_cleanup(client, cluster)
+    request.addfinalizer(fin)
+
+
+def wait_for_cluster_transitioning_message(client, cluster, message):
+    start = time.time()
+    while message not in cluster.transitioningMessage:
+        print(cluster.transitioningMessage)
+        time.sleep(5)
+        cluster = client.reload(cluster)
+        if time.time() - start > MACHINE_TIMEOUT:
+            raise Exception('Timeout waiting for condition')
+    return cluster
+
+
+def wait_until_cluster_is_updating(client, cluster):
+    intermediate_state = "updating"
+    cluster = wait_for_condition(
+        client, cluster,
+        lambda x: x.state == intermediate_state,
+        lambda x: 'State is: ' + x.state,
+        timeout=MACHINE_TIMEOUT)
+    assert cluster.state == intermediate_state
+    return cluster
+
+
+def add_new_etcd_nodes(client, cluster, no_of_nodes=3):
+    node_roles = [["etcd"]] * no_of_nodes
+    aws_nodes = AmazonWebServices().create_multiple_nodes(
+        no_of_nodes, random_test_name(HOST_NAME))
+    i = 0
+    for aws_node in aws_nodes:
+        docker_run_cmd = \
+            get_custom_host_registration_cmd(client, cluster, node_roles[i],
+                                             aws_node)
+        print("Docker run command: " + docker_run_cmd)
+
+        for nr in node_roles[i]:
+            aws_node.roles.append(nr)
+
+        result = aws_node.execute_command(docker_run_cmd)
+        print(result)
+        i += 1
+    return cluster
+
+
+def delete_node_from_ec2(address):
+
+    filters = [
+        {'Name': 'ip-address',
+         'Values': [address]}
+    ]
+    aws_node = AmazonWebServices().get_nodes(filters)
+    AmazonWebServices().delete_node(aws_node[0])
+
+
+def restore_backup_to_recover(client, cluster, b_info):
+    cluster.restoreFromEtcdBackup(etcdBackupId=b_info["backup_id"])
+    validate_cluster(client, cluster, intermediate_state="updating",
```

Comment:
This returns a cluster. which can be return back to the original function ?

Reply:
Not of much use at that stage of the scenario but I will store the returned value.